annotator/__main__.py

```diff
@@ -0,0 +1,440 @@
+    """
+    Finds the name of files in local directory.
+
+    :param path:
+    :param synapse_dir:
+    :param local_root:
+    :param depth:
+    :return: name of file and it's associated parent location/benefactor
+    """
+    path_no_root = path[len(os.path.abspath(local_root)):]
+
+    if depth is not None and path_no_root.count(os.path.sep) > depth - 1:
+
+        if str.startswith(path_no_root, '/'):
+            path_no_root = path_no_root[1:]
+
+        temp_name = path_no_root.split('/')[(depth - 1):]
+        name = '_'.join(temp_name)
+
+        temp_name = '/'.join(temp_name)
+        parent = synapse_dir[os.path.dirname(path[:-len(temp_name)])]
+    else:
+        name = os.path.basename(path)
+        parent = synapse_dir[os.path.dirname(path)]
+
+    return name, parent
+
+
+def create_sync_manifest(file_list, key_list, synapse_dir, local_root, depth):
+    """
+    Creates manifest designed for the input of sync function.
+
+    :param file_list:
+    :param key_list:
+    :param synapse_dir:
+    :param local_root:
+    :param depth:
+    :return:
+    """
+    result = pandas.DataFrame()
+    result['path'] = file_list
+    result['name'] = ""
+    result['parent'] = ""
+
+    for index, row in result.iterrows():
+        row[['name', 'parent']] = _getName(row['path'], synapse_dir, local_root, depth)
+
+    cols = list(result.columns)
+    result = pandas.concat([result, pandas.DataFrame(columns=key_list)])
+    result = result[cols + key_list] # reorder the columns
+    save_path = os.path.join(os.getcwd(),'annotations_manifest.csv')
+    result.to_csv(save_path, index=False)
+    sys.stderr.write('Manifest has been created on local directory: \n %s \n' % os.getcwd())
+
+
+def sync_manifest(args, syn):
+    """
+    Creates a manifest (filepath by annotations) designed for the input of synapse sync
+    function to facilitate file organization and annotations of those files on synapse.
+    
+    :param args:
+    :param syn:
+    :return:
+    """
+    sys.stderr.write('Preparing to create manifest\n')
+    local_root = args.directory
+    synapse_id = args.id
+    annotations = args.files
+    depth = args.n
+
+    if depth is not None:
+        depth = int(depth)
+
+    dir_list, file_list = _getLists(local_root, depth)
+    synapse_dir = _getSynapseDir(syn, synapse_id, local_root, dir_list)
+    key_list = _getAnnotationKey(annotations)
+
+    create_sync_manifest(file_list, key_list, synapse_dir, local_root, depth)
+
+
+def buildParser():
+    """
+    Builds the user-input argument parser.
+
+    :return:
+    """
+    parser = argparse.ArgumentParser()
+
+    subparsers = parser.add_subparsers(title='commands',
+                                       description='The following commands are available:',
+                                       help='For additional help: "annotator <COMMAND> -h"')
+
+    parser_meltjson = subparsers.add_parser('meltjson', help='Creates a flattened synapse table from json files '
```

Comment:
`meltjson` is a bit of a weird term for those of us not in the tidyverse. perhaps use `mergejson` or `combinejson` as an argument?

Reply:
how about json2table? we are doing one more step (un-nesting) after merging/combining.

Reply:
or something more creative... need some sara help here.

Reply:
json2table!